app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java

```diff
@@ -265,15 +265,20 @@ public void notificationSetter(Context context) {
         }
     }
 
-    private void  ArmTimer(Context context, int time) {
+    private void  ArmTimer() {
         Log.e(TAG, "ArmTimer called");
-        if(ActiveBgAlert.getOnly() != null) {
-            Calendar calendar = Calendar.getInstance();
-            AlarmManager alarm = (AlarmManager) getSystemService(ALARM_SERVICE);
-            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.KITKAT) {
-                alarm.setExact(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis() + time, PendingIntent.getService(this, 0, new Intent(this, Notifications.class), 0));
-            } else {
-                alarm.set(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis() + time, PendingIntent.getService(this, 0, new Intent(this, Notifications.class), 0));
+        ActiveBgAlert activeBgAlert = ActiveBgAlert.getOnly();
+        if(activeBgAlert != null ) {
+            AlertType alert = AlertType.get_alert(activeBgAlert.alert_uuid);
+            if(alert != null) {
+                int time = alert.minutes_between;
+                Calendar calendar = Calendar.getInstance();
+                AlarmManager alarm = (AlarmManager) getSystemService(ALARM_SERVICE);
+                if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.KITKAT) {
+                    alarm.setExact(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis() + time, PendingIntent.getService(this, 0, new Intent(this, Notifications.class), 0));
+                } else {
+                    alarm.set(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis() + time, PendingIntent.getService(this, 0, new Intent(this, Notifications.class), 0));
+                }
             }
```

Comment:
since it gets added here https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/blob/master/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/AlertPlayer.java#L56 

wouldnt it basicly behave the same as a snoozed alert until that time has passed?